docs/aws/audit/elbmonitoring/rules/elb_cross_zone_enabled_remediation.mdx

```diff
@@ -1,6 +1,222 @@
+
+1. **Navigate to the EC2 Dashboard:**
+   - Open the AWS Management Console.
+   - In the top search bar, type "EC2" and select "EC2" to go to the EC2 Dashboard.
+
+2. **Access Load Balancers:**
+   - In the left-hand navigation pane, scroll down and click on "Load Balancers" under the "Load Balancing" section.
+
+3. **Select the Load Balancer:**
+   - From the list of load balancers, select the one you want to configure by clicking on its name.
+
+4. **Enable Cross-Zone Load Balancing:**
+   - In the bottom pane, select the "Description" tab.
+   - Click on the "Edit attributes" button.
+   - In the "Edit load balancer attributes" dialog, check the box for "Cross-Zone Load Balancing."
+   - Click "Save" to apply the changes.
+
+By following these steps, you ensure that Cross-Zone Load Balancing is enabled for your ELBs, which helps in distributing traffic evenly across all registered instances in all enabled Availability Zones.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Elastic Load Balancers (ELBs) from having Cross-Zone Load Balancing disabled using AWS CLI, follow these steps:
+
+1. **Identify the Load Balancer:**
+   First, list all your load balancers to identify the one you want to configure.
+
+   ```sh
+   aws elb describe-load-balancers
+   ```
+
+2. **Check Current Cross-Zone Load Balancing Setting:**
+   Verify the current cross-zone load balancing setting for your load balancer.
+
+   ```sh
+   aws elb describe-load-balancer-attributes --load-balancer-name <your-load-balancer-name>
+   ```
+
+3. **Enable Cross-Zone Load Balancing:**
+   Enable cross-zone load balancing for the identified load balancer.
+
+   ```sh
+   aws elb modify-load-balancer-attributes --load-balancer-name <your-load-balancer-name> --load-balancer-attributes "{\"CrossZoneLoadBalancing\":{\"Enabled\":true}}"
+   ```
+
+4. **Verify the Change:**
+   Confirm that cross-zone load balancing has been enabled.
+
+   ```sh
+   aws elb describe-load-balancer-attributes --load-balancer-name <your-load-balancer-name>
+   ```
+
+By following these steps, you can ensure that cross-zone load balancing is enabled for your Elastic Load Balancers using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Elastic Load Balancers (ELBs) from being created without cross-zone load balancing enabled in AWS using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+       aws_secret_access_key='YOUR_AWS_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>